resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -91,6 +91,13 @@ private[spark] class Client(
   private val executorMemoryOverhead = sparkConf.get(EXECUTOR_MEMORY_OVERHEAD).getOrElse(
     math.max((MEMORY_OVERHEAD_FACTOR * executorMemory).toLong, MEMORY_OVERHEAD_MIN)).toInt
 
+  private val isPython = sparkConf.get(IS_PYTHON_APP)
```

Comment:
This is interesting, my one concern here is probably a little esoteric, for mixed language pipelines this might not behave as desired. I'd suggest maybe a JIRA and a note in the config param that it only applies to Python apps not mixed

Reply:
Is there documentation on how to create mixed-language pipelines? Clearly, all you need is a PythonRDD in your plan, but I thought it was non-trivial to create those from a Scala job.

Reply:
It's true, creating mixed language pipelines is difficult and not documented. But I do it, and some others do as well. Some cloud providers (databricks is the most notable example) provide mixed language pipelines in their notebook solutions I believe, and so I think that also reaches a larger audience than the people who do it manually.

Reply:
That's not really documented but as Holden says, it exists. Livy does that - but Livy actually goes ahead and sets the internal `spark.yarn.isPython` property, so it would actually take advantage of this code...

Not sure how others do it, but all the ways I thought on how to expose this as an option were pretty hacky, so I think it's ok to leave things like this for now.

Reply:
Interesting, I'll add this to my example mixed pipeline repo so folks can see this hack.

Reply:
@holdenk, can you point me to that repo? I'd love to have a look at how you do mixed pipelines.

Reply:
Sure, one of them is https://github.com/sparklingpandas/sparklingml